### PR TITLE
chore(webchat): bump react-intl to v3

### PR DESCRIPTION
**Pull request by laurentlp**
This PR upgrades the `react-intl` library to v3 so it is compatible with the component package.

### Screenshots
**en**
Screenshot from 2021-12-01 12-56-19
**fr**
Screenshot from 2021-12-01 12-55-50

### CODE REVIEW

File: packages/webchat/package.json [hunk 1]
```diff
@@ -17,14 +17,14 @@
     "@botpress/messaging-components": "*",
     "@botpress/messaging-socket": "0.0.2",
     "@juggle/resize-observer": "^3.0.2",
+    "@formatjs/intl-pluralrules": "^4.1.6",
+    "@formatjs/intl-utils": "^3.8.4",
```
Comment (laurentlp) [inline packages/webchat/package.json:21]: Those come from the migration guide (https://formatjs.io/docs/react-intl/upgrade-guide-3x/#migrate-to-using-native-intl-apis)

File: packages/webchat/src/components/ConversationList.tsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 import Add from '../icons/Add'
 import { RootStore, StoreDef } from '../store'
 import { RecentConversation } from '../typings'
 
 const ConversationListItem = injectIntl(({ conversation, onClick, hasFocus, intl }: ConversationListItemProps) => {
   const title = intl.formatMessage({ id: 'conversationList.title' }, { id: conversation.id })
-  const date = intl.formatRelative(conversation.lastMessage?.sentOn || conversation.createdOn)
+
+  const { value, unit } = selectUnit(conversation.lastMessage?.sentOn || conversation.createdOn)
+  const date = intl.formatRelativeTime(value, unit)
```
Comment (laurentlp) [inline packages/webchat/src/components/ConversationList.tsx:14]: Same here. See point .6 (https://formatjs.io/docs/react-intl/upgrade-guide-3x/#formattedrelativetime)

File: packages/webchat/src/components/Composer.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { observe } from 'mobx'
 import { inject, observer } from 'mobx-react'
 import React from 'react'
-import { FormattedMessage, InjectedIntlProps, injectIntl } from 'react-intl'
+import { WrappedComponentProps, injectIntl } from 'react-intl'
```
Comment (laurentlp) [inline packages/webchat/src/components/Composer.tsx:4]: `InjectedIntlProps` was renamed to `WrappedComponentProps`. See info buble: https://formatjs.io/docs/react-intl/upgrade-guide-3x/#typescript-support

File: packages/webchat/src/main.tsx [hunk 4]
```diff
@@ -416,40 +416,40 @@ export default inject(({ store }: { store: RootStore }) => ({
...
-  | 'updateConfig'
-  | 'mergeConfig'
-  | 'isWebchatReady'
-  | 'incrementUnread'
-  | 'displayWidgetView'
-  | 'resetUnread'
-  | 'setLoadingCompleted'
-  | 'dimensions'
->
+type MainProps = { store: RootStore } & WrappedComponentProps &
```
Comment (laurentlp) [inline packages/webchat/src/main.tsx:421]: Adding `& WrappedComponentProps` here allows removing the `@ts-ingore` at line 419

File: packages/webchat/src/store/index.ts [hunk 5]
```diff
@@ -59,7 +57,7 @@ class RootStore {
   @observable
   public messageFeedbacks!: EventFeedback[]
 
-  public intl!: InjectedIntl
+  public intl!: IntlShape
```
Comment (laurentlp) [inline packages/webchat/src/store/index.ts:60]: `InjectedIntl` was renamed to `IntlShape`

File: packages/webchat/src/translations/index.tsx [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+require('@formatjs/intl-pluralrules/polyfill')
+require('@formatjs/intl-pluralrules/locale-data/ar')
+require('@formatjs/intl-pluralrules/locale-data/de')
+require('@formatjs/intl-pluralrules/locale-data/en')
+require('@formatjs/intl-pluralrules/locale-data/es')
+require('@formatjs/intl-pluralrules/locale-data/fr')
+require('@formatjs/intl-pluralrules/locale-data/it')
+require('@formatjs/intl-pluralrules/locale-data/pt')
+require('@formatjs/intl-pluralrules/locale-data/ru')
+require('@formatjs/intl-pluralrules/locale-data/uk')
```
Comment (laurentlp) [inline packages/webchat/src/translations/index.tsx:10]: See: https://formatjs.io/docs/react-intl/upgrade-guide-3x/#migrate-to-using-native-intl-apis

**Review by samuelmasse — APPROVED** [review]
LGTM

**Pull request merged**